lib/Retriever.js

```diff
@@ -93,11 +96,12 @@ class Retriever {
                                 this.templateDir,
                                 name,
                                 templateVariables,
-                                subType
+                                subTypeArr?.[0] // this might cause issues if one were to try to use the method with multiple subtypes
```

Comment:
make sure comment is clear what objective is here

Reply:
added warning if more than 1 subtype was passed to retrieveAsTemplate: `retrieveAsTemplate only works with one subtype, ignoring all but first subtype from your list: ${subTypeArr.join(', ')`